securesystemslib/hsm.py

```diff
@@ -23,29 +23,40 @@
 from __future__ import division
 from __future__ import unicode_literals
 
-# Import python wrapper for PKCS#11 to communicate with the tokens
-import PyKCS11
-
 import binascii
 import logging
 import securesystemslib.exceptions
 
 # Import cryptography routines needed to retrieve cryptographic
-# keys and certificates in PEM format.
-from cryptography.hazmat.primitives import serialization
-from cryptography.hazmat.backends import default_backend
-from cryptography import x509
-
-# RSA-PSS with SHA256 hash to be used for signature generation.
-RSA_PSS_MECH = PyKCS11.CKM_SHA256_RSA_PKCS_PSS
-# SHA256 hash to be used to digest the data.
-RSA_PSS_HASH_SHA256 = PyKCS11.CKM_SHA256
-# Mask generating function for SHA256 Hash.
-RSA_PSS_MGF_SHA256 = PyKCS11.CKG_MGF1_SHA256
-# Length of salt to be used for hashing.
-RSA_PSS_SALT_LENGTH = 32
-
-logger = logging.getLogger('securesystemslib_hsm')
+# keys and certificates.
+CRYPTO = True
+NO_CRYPTO_MSG = "To retrieve cryptographic keys and certificates " \
+                "cryptography library is needed"
+try:
+  from cryptography.hazmat.primitives import serialization
+  from cryptography.hazmat.backends import default_backend
+  from cryptography import x509
+except ImportError:
+  CRYPTO = False
+
+# Import python wrapper for PKCS#11 to communicate with the tokens
+HSM_SUPPORT = True
+NO_HSM_MSG = "HSM support required PyKCS11 library"
+try:
+  import PyKCS11
+
+  # RSA-PSS with SHA256 hash to be used for signature generation.
+  RSA_PSS_MECH = PyKCS11.CKM_SHA256_RSA_PKCS_PSS
+  # SHA256 hash to be used to digest the data.
+  RSA_PSS_HASH_SHA256 = PyKCS11.CKM_SHA256
+  # Mask generating function for SHA256 Hash.
+  RSA_PSS_MGF_SHA256 = PyKCS11.CKG_MGF1_SHA256
+  # Length of salt to be used for hashing.
+  RSA_PSS_SALT_LENGTH = 32
+except ImportError:
+  HSM_SUPPORT = False
+
+logger = logging.getLogger('__securesystemslib_hsm__')
```

Comment:
```suggestion
logger = logging.getLogger(__name__)
```
There was a recent change in 5c1a1d953cb00b481d277e6f48c5003d1c9e9d0c to standardise on the pattern `logger = logging.getLogger(__name__)`